rcl/include/rcl/node.h

```diff
@@ -124,17 +144,24 @@ rcl_get_zero_initialized_node(void);
  *
  * \param[inout] node a preallocated rcl_node_t
  * \param[in] name the name of the node
+ * \param[in] name_space the namespace of the node
  * \param[in] options the node options
  * \return `RCL_RET_OK` if the node was initialized successfully, or
  * \return `RCL_RET_ALREADY_INIT` if the node has already be initialized, or
  * \return `RCL_RET_INVALID_ARGUMENT` if any arguments are invalid, or
  * \return `RCL_RET_BAD_ALLOC` if allocating memory failed, or
+ * \return `RCL_RET_NODE_INVALID_NAME` if the name is invalid, or
+ * \return `RCL_RET_NODE_INVALID_NAMESPACE` if the name_space is invalid, or
  * \return `RCL_RET_ERROR` if an unspecified error occurs.
  */
 RCL_PUBLIC
 RCL_WARN_UNUSED
 rcl_ret_t
-rcl_node_init(rcl_node_t * node, const char * name, const rcl_node_options_t * options);
+rcl_node_init(
+  rcl_node_t * node,
+  const char * name,
+  const char * name_space,
```

Comment:
I had two reasons for placing it like this.

The first was because I believe the namespace should be of elevated importance (similar to node name). Your same argument could be made for `name`, specifically that it should also be in the options struct, but to emulate the interface of ROS 1, node name and node namespace are high level concepts.

On the technical side, it is nice that the options structure remains a POD. As soon as I add a pointer (like a `const char *`) it is no longer "trivially copyable" (borrowing a term from C++). 

That all being said, I don't feel strongly about it. I definitely agree that most options should end up in the structure, but in this instance I made a decision to leave it out.